lib/collection.js

```diff
@@ -2101,6 +2101,12 @@ Collection.prototype.findOneAndUpdate = function(filter, update, options, callba
   if (typeof options === 'function') (callback = options), (options = {});
   options = options || {};
 
+  var err = checkForAtomicOperators(update);
```

Comment:
1. Use `const` instead of `var` here.
2. Move the check to after the `if (update == null || typeof update !== 'object')`.